src/uu/split/src/split.rs

```diff
@@ -125,28 +127,50 @@ pub fn uu_app() -> Command {
         )
         .arg(
             Arg::new(OPT_NUMERIC_SUFFIXES)
-                .short('d')
                 .long(OPT_NUMERIC_SUFFIXES)
+                .require_equals(true)
                 .default_missing_value("0")
                 .num_args(0..=1)
+                .overrides_with(OPT_NUMERIC_SUFFIXES)
                 .help("use numeric suffixes instead of alphabetic"),
         )
         .arg(
-            Arg::new(OPT_SUFFIX_LENGTH)
-                .short('a')
-                .long(OPT_SUFFIX_LENGTH)
-                .value_name("N")
-                .default_value(OPT_DEFAULT_SUFFIX_LENGTH)
-                .help("use suffixes of fixed length N. 0 implies dynamic length."),
+            Arg::new(OPT_NUMERIC_SUFFIXES_SHORT)
+                .short('d')
+                .action(clap::ArgAction::SetTrue)
+                .value_parser(|_: &str| -> Result<String,clap::Error> {
+                    Ok("0".to_string()) // force value to "0"
+                })
+                .overrides_with(OPT_NUMERIC_SUFFIXES_SHORT)
+                .help("use numeric suffixes instead of alphabetic"),
         )
         .arg(
             Arg::new(OPT_HEX_SUFFIXES)
-                .short('x')
                 .long(OPT_HEX_SUFFIXES)
                 .default_missing_value("0")
+                .require_equals(true)
                 .num_args(0..=1)
+                .overrides_with(OPT_HEX_SUFFIXES)
+                .help("use hex suffixes instead of alphabetic"),
+        )
+        .arg(
+            Arg::new(OPT_HEX_SUFFIXES_SHORT)
+                .short('x')
+                .action(clap::ArgAction::SetTrue)
+                .value_parser(|_: &str| -> Result<String,clap::Error> {
+                    Ok("0".to_string()) // force value to "0"
+                })
```

Comment:
This feels convoluted. We have have to combine the arguments anyway, so it would suffice for this to just be a flag without any value.

Reply:
well ... it is a flag as far as behaviour goes (and as far as clap works), but it does produce a value. If -d or -x is used , the value must be specifically set to 0 for numeric or hex suffix correspondingly. It can be set here with `value_parser()` or harcoded later on when suffixes are processed. Would it be better to have all defaults in the same place as arg definitions?

Reply:
I see where you're coming from. It's an interesting idea to try to make all the flags sort of the same in the code. I think I still prefer hardcoding them in the processing of the flags. The thing that I find confusing is that if it is a flag, we should use the functions that are appropriate for flags if that makes sense. `value_parser` is usually used for parsing user generated values, which we don't have here, so hardcoding the `0` gives us the ability to skip the parsing.

So, for instance, I'd write the match expression to combine the flags something more like this:
```rust
 match (
    matches.get_one::<String>(OPT_NUMERIC_SUFFIXES),
    matches.get_one::<String>(OPT_HEX_SUFFIXES),
    matches.get_flag(OPT_NUMERIC_SUFFIXES_SHORT),
    matches.get_flag(OPT_HEX_SUFFIXES_SHORT),
) {
    (Some(v), _, _, _) => parse_num_suffix(matches, v),
    (_, Some(v), _, _) => parse_hex_suffix(matches, v),
    (_, _, true, _) => Ok((SuffixType::Decimal, 0)),
    (_, _, _, true) => Ok((SuffixType::Hexadecimal, 0)),
    _ => Ok((SuffixType::Alphabetic, 0)), // no numeric/hex suffix, using default alphabetic
}
```
(I forgot the exact semantics of those `clap` functions. Maybe the value source check is necessary as well)

Reply:
refactored to remove arg `.value_parser()` and hardcoded default `0` in the `fn suffix_type_from()`